subprojects/core/src/main/java/org/gradle/api/internal/SettingsInternal.java

```diff
@@ -67,6 +68,8 @@
 
     File getBuildSrcDir();
 
+    ServiceRegistry getServices();
```

Comment:
Precompiled Groovy settings plugins were using the wrong services scope - GradleScopeServices - which should be used by init scripts. Exposing the settings service registry here allows precompiled settings plugins to fetch the right one.